controllers/oas_parser.go

```diff
@@ -0,0 +1,41 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/buger/jsonparser"
+)
+
+var ClientCertificatePath = []string{TykOASExtenstionStr, "server", "clientCertificates", "allowlist"}
+
+func OASSetClientCertificates(data []byte, certid string) ([]byte, error) {
+	allowlist := []string{}
+
+	val, _, _, err := jsonparser.Get(data, ClientCertificatePath...)
+	if err != nil && err != jsonparser.KeyPathNotFoundError {
+		return nil, err
+	}
+
```

Comment:
let's move the creation of `allowList` below the following `if` condition. If the `if` statement fails, we don't need `allowList`, and creating it would be unnecessary.

```suggestion
	val, _, _, err := jsonparser.Get(data, ClientCertificatePath...)
	if err != nil && err != jsonparser.KeyPathNotFoundError {
		return nil, err
	}
	
	allowlist := []string{}

```